### PR TITLE
Add tests for typed modal

### PR DESCRIPTION
**Pull request by alschmiedt**

### CODE REVIEW

File: packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+/**
+ * @fileoverview Unit tests for TypedVariableModal.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+const assert = require('assert');
+const Blockly = require('blockly');
+const sinon = require('sinon');
+
+const TypedVariableModal = require('../dist/index.js').TypedVariableModal;
```
Comment (samelhusseini) [inline packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js:16]: This is not a this PR question but in general, why are we including the dist version (rather than including src/index.js)
Avoiding the need for a build before running these tests.

Comment (alschmiedt) [reply packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js:16]: So just tried to change this to src/index.js and I think the answer is that we need the umd build in order to run in node.

Comment (samelhusseini) [reply packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js:16]: Ah makes sense.

File: packages/plugins/typed-variable-modal/src/TypedVariableModal.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ export class TypedVariableModal extends Modal {
     /**
      * An array holding arrays with the name of the type and the display name
      *     for the type. Ex: [['Penguin', 'PENGUIN'], ['Giraffe', 'GIRAFFE']].
-     * @type {Array<Array<string>>}
+     * @type {Array.<Array.<string>>}
```
Comment (moniika) [inline packages/plugins/typed-variable-modal/src/TypedVariableModal.js]: Why was this changed? Based on [this](https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System) it seems like the "." is deprecated syntax.

File: packages/plugins/typed-variable-modal/src/TypedVariableModal.js [hunk 3]
```diff
@@ -238,7 +238,7 @@ export class TypedVariableModal extends Modal {
   renderContent_(contentContainer) {
     const varNameContainer = this.createVarNameContainer_();
     this.variableNameInput_ = varNameContainer
-        .querySelector('.typed-modal-variable-name-input');
+        .querySelector('.typedModalVariableNameInput');
```
Comment (moniika) [inline packages/plugins/typed-variable-modal/src/TypedVariableModal.js:241]: Just checking, but this change is so that we are following pattern seen in Blockly core right?

File: packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+  teardown(() => {
+    this.jsdomCleanup();
+    sinon.restore();
+  });
+
+  suite('init()', () => {
+    test('Render is called', () => {
+      this.workspace.registerButtonCallback = sinon.fake();
+      this.typedVarModal.init();
+      assert(this.workspace.registerButtonCallback.calledOnce);
```
Comment (moniika) [inline packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js]: Roughly looking at documentation, it seems like you could do:
`sinon.assert.calledOnce(this.workspace.registerButtonCallback);`
Assuming it works, I think it might give better error messages.

File: packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+        'CREATE_TYPED_VARIABLE', types);
+  });
+
+  teardown(() => {
+    this.jsdomCleanup();
+    sinon.restore();
+  });
+
+  suite('init()', () => {
+    test('Render is called', () => {
```
Comment (moniika) [inline packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js]: a nit, the other test name omit the "be" verb (so "is" in this case)

File: packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+    test('Events and button callback removed', () => {
+      this.typedVarModal.init();
+      this.workspace.removeButtonCallback = sinon.fake();
+      const numEvents = this.typedVarModal.boundEvents_.length;
+      Blockly.unbindEvent_ = sinon.fake();
+      this.typedVarModal.dispose();
+
+      assert(this.workspace.removeButtonCallback.calledOnce);
+      assert.equal(document.querySelector('.blocklyModalOverlay'), null);
+      assert.equal(Blockly.unbindEvent_.callCount, numEvents);
```
Comment (moniika) [inline packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js]: Once again, not sure if this works, but looking [here](https://sinonjs.org/releases/latest/assertions/) I also found `sinon.assert.callCount`

File: packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+      this.typedVarModal.getSelectedType_ = sinon.fake.returns('Giraffe');
+      this.typedVarModal.getDisplayName_ = sinon.fake.returns('Giraffe');
+    });
+    test('No text', () => {
+      this.typedVarModal.getValidInput_ = sinon.fake.returns(null);
+      this.typedVarModal.onConfirm_();
+      assert(Blockly.alert
+          .calledWith('Name is not valid. Please choose a different name.'));
+    });
+    test('Valid Name', () => {
```
Comment (moniika) [inline packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js]: nit: Name > name

File: packages/plugins/typed-variable-modal/src/TypedVariableModal.js (hunk 2 above)
Comment (alschmiedt) [reply packages/plugins/typed-variable-modal/src/TypedVariableModal.js]: Good catch. This was in my attempt to figure out what the heck was happening with my tests. Reverted.

File: packages/plugins/typed-variable-modal/src/TypedVariableModal.js (hunk 3 above)
Comment (alschmiedt) [reply packages/plugins/typed-variable-modal/src/TypedVariableModal.js:241]: Yup, the rest of the names were changed but I forgot about this one.

File: packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js (hunk 5 above)
Comment (alschmiedt) [reply packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js]: done

File: packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js (hunk 4 above)
Comment (alschmiedt) [reply packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js]: done

File: packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js (hunk 6 above)
Comment (alschmiedt) [reply packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js]: done

File: packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js (hunk 7 above)
Comment (alschmiedt) [reply packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js]: done

**Review by moniika — APPROVED** [review]
